x/axelarnet/types/evm_translator.go

```diff
@@ -145,6 +161,44 @@ func ConstructWasmMessage(gm nexus.GeneralMessage, payload []byte) ([]byte, erro
 	return json.Marshal(msg)
 }
 
+// ConstructWasmMessageV2 creates a json serialized wasm message from json encoded payload
+func ConstructWasmMessageV2(gm nexus.GeneralMessage, payload []byte) ([]byte, error) {
+	executeMsg := make(map[string]interface{})
+	err := json.Unmarshal(payload, &executeMsg)
+	if err != nil {
+		return nil, err
+	}
+	// json payload must have single key, the method name
+	if len(executeMsg) != 1 {
+		return nil, fmt.Errorf("invalid payload")
+	}
+
+	for _, msg := range executeMsg {
```

Comment:
What does it mean for the `executeMsg` here to contain multiple method names?

Reply:
the destinatio ibc hook middlewre won't execute the message, and fail the tx

Reply:
Then don't we wanna validate this somewhere?

Reply:
oh we do, on line 172, there is a check that it can have one key, val pair

Reply:
maybe add a comment? i believe this was done since you don't know the key/method name to index into